packages/contracts-bedrock/src/L2/L1BlockHolocene.sol

```diff
@@ -0,0 +1,120 @@
+        } else if (_type == ConfigType.REMOVE_DEPENDENCY) {
+            _removeDependency(_value);
+        } else if (_type == ConfigType.SET_BATCHER_HASH) {
+            _setBatcherHash(_value);
+        } else if (_type == ConfigType.SET_GAS_CONFIG_ECOTONE) {
+            _setGasConfigEcotone(_value);
+        } else if (_type == ConfigType.SET_GAS_LIMIT) {
+            _setGasLimit(_value);
+        }
+    }
+
+    /// @notice Internal method to set new batcher hash.
+    /// @param _value The encoded value with which to set the new batcher hash.
+    function _setBatcherHash(bytes calldata _value) internal {
+        bytes32 _batcherHash = StaticConfig.decodeSetBatcherHash(_value);
+
+        batcherHash = _batcherHash;
+
+        emit BatcherHashSet(_batcherHash);
+    }
+
+    /// @notice Internal method to set new gas config.
+    /// @param _value The encoded value with which to set the new batcher hash.
+    function _setGasConfigEcotone(bytes calldata _value) internal {
+        uint256 _scalar = StaticConfig.decodeSetGasConfigEcotone(_value);
+
+        (uint32 _blobBaseFeeScalar, uint32 _baseFeeScalar) = _decodeScalar(_scalar);
+
+        blobBaseFeeScalar = _blobBaseFeeScalar;
+        baseFeeScalar = _baseFeeScalar;
+
+        emit GasConfigEcotoneSet(_blobBaseFeeScalar, _baseFeeScalar);
+    }
+
+    /// @notice Internal method to decode blobBaseFeeScalar and baseFeeScalar.
+    /// @return Decoded blobBaseFeeScalar and baseFeeScalar.
+    function _decodeScalar(uint256 _scalar) internal pure returns (uint32, uint32) {
+        // _scalar is constructed as follows:
+        // uint256 _scalar = (uint256(0x01) << 248) | (uint256(_blobbasefeeScalar) << 32) | _basefeeScalar;
+        // where _blobbasefeeScalar and _basefeeScalar are both uint32.
+
+        require(0x01 == _scalar >> 248, "invalid _scalar");
+
+        uint32 _blobbasefeeScalar = uint32((_scalar >> 32) & 0xffffffff);
+        uint32 _basefeeScalar = uint32(_scalar & 0xffffffff);
+        return (_blobbasefeeScalar, _basefeeScalar);
+    }
+
+    /// @notice Internal method to set new gas limit.
+    /// @param _value The encoded value with which to set the new gas limit.
+    function _setGasLimit(bytes calldata _value) internal {
+        uint64 _gasLimit = StaticConfig.decodeSetGasLimit(_value);
+        gasLimit = _gasLimit;
+        emit GasLimitSet(_gasLimit);
+    }
+
+    /// @notice Updates the L1 block values for an Holocene upgraded chain.
+    /// Params are packed and passed in as raw msg.data instead of ABI to reduce calldata size.
+    /// Params are expected to be in the following order:
+    ///   1. _sequenceNumber     Number of L2 blocks since epoch start.
+    ///   2. _timestamp          L1 timestamp.
+    ///   3. _number             L1 blocknumber.
+    ///   4. _basefee            L1 base fee.
+    ///   5. _blobBaseFee        L1 blob base fee.
+    ///   6. _hash               L1 blockhash.
+    function setL1BlockValuesHolocene() external {
```

Comment:
We will want to implement the following comment as `setL1BlockValuesHolocene`: https://github.com/ethereum-optimism/specs/issues/122#issuecomment-2293704907

Reply:
Done in https://github.com/ethereum-optimism/optimism/pull/11600/commits/4618303e2fde6a390e168b08e8a2cdb3f7661b86